## 2017년 카카오 신입 공채 블라인드 시험문제 1번( 비밀지도 알고리즘: 난이도 하)

링크 : https://programmers.co.kr/learn/courses/30/lessons/17681?language=python3

## ■ 카카오 비밀지도 문제를 풀기 위한 기본 문제 7개들

## ■ 1. Bin

" Bin -> 10진수를 2진수로 바꾸어주는 코드 "

### 문제1. 숫자 9를 이진수로 변경하시오!

In [1]:
print(bin(9))
print(type(bin(9))) # type은 str로 결과값은 문자열임.
print(bin(9)[2:]) # 0b 부분을 잘라내기 위해서 [2:]를 사용한다.

0b1001
<class 'str'>
1001


## ■ 2. format 함수

" 이 또한 10진수를 2진수로 바꾸어주는 코드 "

In [2]:
result = format(9,'b') # 9는 '변수명'자리 (int형만 가능), 'b' = binary(2진법)
print(result) # 1001로 0b없이 깔끔하게 나옴.
print(type(result)) # type은 str로 결과값은 문자열임.

1001
<class 'str'>


## ■ 3. rjust 함수(오른쪽 정렬)

" 현재 문제에서는 '1001' 4자리 이진수를 '01001' 5자리 이진수로 만들어 내기위하여 사용된다. "

In [5]:
print(bin(9)[2:].rjust(5,'0')) 

# rjust = 오른쪽 정렬
# 5 = 문자열크기를 '5'로 정함
# 나머지자리는 0으로 채움.

print(bin(11)[2:].rjust(7,'0'))

01001
0001011


## ■ 4. zfill 함수(파이썬 문자열 앞으로 0채우기)

" 현재 문제에서는 '1001' 4자리 이진수를 '01001' 5자리 이진수로 만들어 내기위하여 사용된다. "

In [ ]:
result = bin(9)[2:]

print(result)
print(result.zfill(5))

### 문제2. (카카오 비밀지도 문제) 숫자 30을 이진수로 아래와 같이 출력하시오! 

n = 5  
a = bin(9)  
b = bin(30)  

print(a,b) # 0b1001, 0b11110  
print(change(a,n), change(b,n)) # 01001, 11110

In [6]:
# rjust로 풀기

def change(x,n):
    return x[2:].rjust(n,'0')

n = 5
a = bin(9)
b = bin(30)

print(a,b) # 0b1001, 0b11110
print(change(a,n), change(b,n)) # 01001, 11110

0b1001 0b11110
01001 11110


In [ ]:
# zfill로 풀기

def change(x,n):
    return x[2:].zfill(n)

n = 5
a = bin(9)
b = bin(30)

print(a,b) # 0b1001, 0b11110
print(change(a,n), change(b,n)) # 01001, 11110

### 문제5. 위의 만든 change 함수를 이용해서 아래의 두개의 리스트를 가지고 아래의 결과를 출력하시오!

n = 5  
arr1 = [9,20,28,18,11]  
arr2 = [30,1,21,17,28]  

결과  
01001, 11110  
10100, 00001  
11100, 10101  
10010, 10001  
01011, 11100  

In [ ]:
def change(x,n):
    return x[2:].rjust(n,'0')

n = 5
arr1 = [9,20,28,18,11]
arr2 = [30,1,21,17,28]

for i,j in zip(arr1,arr2):
    print (change(bin(i),5), change(bin(j),5))

## ■ 문자열 합치기

In [22]:
str1 = '#'
str2 = ' '

print(str1+str1+str2+str1)

## #
